### ISSUE TITLE
paragraph.dart  _cachedRect is not refreshed after the window width changes, which results in limited selection

### ISSUE THREAD

**Issue by yuuuuuzj** [flutter/flutter#155143]
### Steps to reproduce

After the first rendering, the selected file can be selected. After reducing the window size, some selected areas are limited and cannot be selected

### Expected results

It is expected that after the window width is adjusted, the selection can also be normal, and the area is not limited.

### Actual results

After the first rendering, the selected file can be selected. After reducing the window size, some selected areas are limited and cannot be selected

### Code sample

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: Text('Selectable Text with Dynamic Layout'),
        ),
        body: const DynamicLayoutWidget(),
      ),
    );
  }
}

class DynamicLayoutWidget extends StatefulWidget {
  const DynamicLayoutWidget({Key? key}) : super(key: key);

  @override
  _DynamicLayoutWidgetState createState() => _DynamicLayoutWidgetState();
}

class _DynamicLayoutWidgetState extends State<DynamicLayoutWidget> {
  @override
  Widget build(BuildContext context) {
    return const Material(
      color: Colors.transparent,
      child: SelectionArea(
          child: Text(
        'This is a selectable text.Selectable Text with WrapSelectable Text with WrapSelectable Text with WrapSelectable Text with WrapSelectable Text with Wrap',
        style: TextStyle(fontSize: 18),
        selectionColor: Colors.blue,
      )),
    );
  }
}
```
</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/eda9de0a-79e9-47b7-a79f-4a96e20e6e73

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

Flutter assets will be downloaded from https://storage.flutter-io.cn. Make sure you trust this source!
[√] Flutter (Channel stable, 3.24.1, on Microsoft Windows [版本 10.0.22631.4169], locale zh-CN)
    • Flutter version 3.24.1 on channel stable at D:\flutter_windows_3.24.1-stable\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5874a72aa4 (3 weeks ago), 2024-08-20 16:46:00 -0500
    • Engine revision c9b9d5780d
    • Dart version 3.5.1
    • DevTools version 2.37.2
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[√] Windows Version (Installed version of Windows is version 10 or higher)

[X] Android toolchain - develop for Android devices
    X Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/to/windows-android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.10.2)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.10.35004.147
    • Windows 10 SDK version 10.0.22621.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/to/windows-android-setup for detailed instructions).

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [版本 10.0.22631.4169]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 128.0.6613.138
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 128.0.2739.67

[√] Network resources
    • All expected network resources are available.

! Doctor found issues in 2 categories.

**Comment by danagbemava-nc** [comment]
Reproducible using the code sample provided above on the latest stable & master. 

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 3.24.3, on macOS 14.6.1 23G93 darwin-arm64, locale en-US)
    • Flutter version 3.24.3 on channel stable at /Users/deanli/dev/stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2663184aa7 (34 hours ago), 2024-09-11 16:27:48 -0500
    • Engine revision 36335019a8
    • Dart version 3.5.3
    • DevTools version 2.37.3
```

```
[!] Flutter (Channel master, 3.26.0-1.0.pre.113, on macOS 14.6.1 23G93 darwin-arm64, locale en-US)
    • Flutter version 3.26.0-1.0.pre.113 on channel master at /Users/deanli/dev/master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 1d5f36f014 (5 hours ago), 2024-09-12 22:59:31 -0400
    • Engine revision 04802b7790
    • Dart version 3.6.0 (build 3.6.0-248.0.dev)
    • DevTools version 2.40.0-dev.1
```
</details>

**Comment by yuuuuuzj** [comment]
> Reproducible using the code sample provided above on the latest stable & master.
> 
> flutter doctor -v

I checked the code for 24.1 and made no adjustments. Please pay more attention to the desktop experience

**Comment by yuuuuuzj** [comment]
Is there any conclusion on this question

**Comment by Renzo-Olivares** [comment]
Hi @yuuuuuzj, thank you for reporting this issue. I have a fix out in review #155719.
